datasetinsights/io/downloader/http_downloader.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import os
+import tempfile
+import zipfile
+
+from datasetinsights.io.download import download_file, validate_checksum
+from datasetinsights.io.downloader.base import DatasetDownloader
+from datasetinsights.io.exceptions import ChecksumError
+
+logger = logging.getLogger(__name__)
+
+DATASETS = ["synthetic", "groceries_real"]
+
+
+class HTTPDownloader(DatasetDownloader, protocol="http://"):
+    def download(self, source_uri, output, checksum_file=None, **kwargs):
+
+        dataset_path = os.path.join(output, "dataset.zip")
+        download_file(source_uri, dataset_path)
+
+        if checksum_file:
+
+            checksum = HTTPDownloader.get_checksum_from_file(checksum_file)
+
+            try:
+                validate_checksum(dataset_path, checksum)
+            except ChecksumError as e:
+                logger.info("Checksum mismatch. Delete the downloaded file.")
+                os.remove(dataset_path)
+                raise e
+
+        HTTPDownloader.unzip_file(dataset_path, output)
+
+    @staticmethod
+    def get_checksum_from_file(filepath):
+
+        if filepath.startswith(("http://", "https://")):
+            with tempfile.TemporaryDirectory() as tmp:
+                checksum_file_path = os.path.join(tmp, "checksum.txt")
+                download_file(source_uri=filepath, dest_path=checksum_file_path)
+                return HTTPDownloader.read_checksum_from_txt(checksum_file_path)
+
+        elif filepath.startswith("/"):
```

Comment:
You might want to do `os.path.isfile(fname) ` instead. It's possible that the user specifies a relative path to the file based on where the user is executing the command.

Reply:
Thanks, will do.